crates/router/src/db/reverse_lookup.rs

```diff
@@ -17,154 +25,76 @@ pub trait ReverseLookupInterface {
-                        sk_id: new.sk_id.clone(),
-                        pk_id: new.pk_id.clone(),
-                        source: new.source.clone(),
-                        updated_by: storage_scheme.to_string(),
-                    };
-                    let redis_entry = kv::TypedSql {
-                        op: kv::DBOperation::Insert {
-                            insertable: kv::Insertable::ReverseLookUp(new),
-                        },
-                    };
+        let redis_fut = redis_conn.get_and_deserialize_key(key, "ReverseLookup");
 
-                    match kv_wrapper::<ReverseLookup, _, _>(
-                        self,
-                        KvOperation::SetNx(&created_rev_lookup, redis_entry),
-                        format!("reverse_lookup_{}", &created_rev_lookup.lookup_id),
-                    )
-                    .await
-                    .map_err(|err| err.to_redis_failed_response(&created_rev_lookup.lookup_id))?
-                    .try_into_setnx()
-                    {
-                        Ok(SetnxReply::KeySet) => Ok(created_rev_lookup),
-                        Ok(SetnxReply::KeyNotSet) => Err(errors::StorageError::DuplicateValue {
-                            entity: "reverse_lookup",
-                            key: Some(created_rev_lookup.lookup_id.clone()),
-                        })
-                        .into_report(),
-                        Err(er) => Err(er).change_context(errors::StorageError::KVError),
-                    }
-                }
-            }
-        }
-
-        #[instrument(skip_all)]
-        async fn get_lookup_by_lookup_id(
-            &self,
-            id: &str,
-            storage_scheme: enums::MerchantStorageScheme,
-        ) -> CustomResult<ReverseLookup, errors::StorageError> {
-            let database_call = || async {
-                let conn = connection::pg_connection_read(self).await?;
-                ReverseLookup::find_by_lookup_id(id, &conn)
-                    .await
-                    .map_err(Into::into)
-                    .into_report()
-            };
-
-            match storage_scheme {
-                enums::MerchantStorageScheme::PostgresOnly => database_call().await,
-                enums::MerchantStorageScheme::RedisKv => {
-                    let redis_fut = async {
-                        kv_wrapper(
-                            self,
-                            KvOperation::<ReverseLookup>::Get,
-                            format!("reverse_lookup_{id}"),
-                        )
-                        .await?
-                        .try_into_get()
-                    };
-
-                    Box::pin(db_utils::try_redis_get_else_try_database_get(
-                        redis_fut,
-                        database_call,
-                    ))
-                    .await
-                }
-            }
-        }
+        Box::pin(try_redis_get_else_try_database_get(
```

Comment:
why we falling back to database here?

Reply:
This was there to ensure smooth transition of old data. But have removed this since KV is not enabled on all envs.